Feat/neo version

Noxaelum opened the pull request.

Noxaelum committed:
Initialize shadcn to project

Noxaelum committed:
Move app folder inside src folder

Noxaelum committed:
Update config files to reflect src folder change

Noxaelum committed:
Update shadcn import folders

Noxaelum committed:
Add shadcn button and sheet components with modifications

Noxaelum committed:
Add nlec colors

Noxaelum committed:
Add header image

Noxaelum committed:
Create fonts asset file

Noxaelum committed:
Create navbar organism and its molecules

Noxaelum committed:
Create header section

Noxaelum committed:
Add scroll-smooth css
Edit main element to have proper top padding

Noxaelum committed:
Add HeaderSection to Home page

Noxaelum committed:
Add about-image

Noxaelum committed:
Create about-section component

Noxaelum committed:
Add about-section to home page

Noxaelum committed:
Add scroll-padding-top and remove height of 3000

Noxaelum committed:
Add active psuedo-element to button
Add margin-bottom to header
Add SheetClose to mobile menu links

Noxaelum committed:
Create ServicesSection component
Add shadcn card and modify it

Noxaelum committed:
Add responsive margin-bottom to about section
Use SectionTitle on About Us header
Remove type on Navbar

Noxaelum committed:
Add ServicesSection to Home page

Noxaelum committed:
Create SectionTitle atom
Add proper type to items prop on NavbarLinks

Noxaelum committed:
Add images and icons for impact and footer components

Noxaelum committed:
Create content-wrapper and section-wrapper

Noxaelum committed:
Create footer, impact-section, and join-us-section

Noxaelum committed:
Add Footer, impact and join-us sections to layout and page files respectively

Noxaelum committed:
Add better spacing to components
Use section and content wrapper in components
Add none size option to button

Noxaelum committed:
Add nvmrc file

Noxaelum committed:
Merge branch 'feat/initial-version' into feat/neo-version

Noxaelum committed:
Update header section to reflect neo version

Noxaelum committed:
Update about-section, section-title, and section-wrapper

Noxaelum committed:
Finish service section. Partially finish what-we-do-section

Noxaelum committed:
Update impact-section

Noxaelum committed:
Install required packages

Noxaelum committed:
Add toaster for informing users successful submits
Add contact us form
Update footer, impact-section, join-us-section, and what-we-do-section

Noxaelum committed:
Update code to fix build errors

idiglove commented on src/app/layout.tsx:
```diff
@@ -0,0 +1,32 @@
+import type { Metadata } from "next";
+import "./globals.css";
+// import Navbar from "@/components/organisms/navbar";
```
remove comments pls

idiglove commented on src/app/layout.tsx:
```diff
@@ -0,0 +1,32 @@
+
+export const metadata: Metadata = {
+  title: "NLEC app",
+  description: "",
```
ask Engramar for seo metadata

idiglove commented on src/app/layout.tsx:
```diff
@@ -0,0 +1,32 @@
+  return (
+    <html
+      lang="en"
+      // className="scroll-pt-[calc(65px+1rem)] scroll-smooth md:scroll-pt-[calc(69px+2rem)] xl:scroll-pt-[calc(73px+3rem)]"
```
remove comments

idiglove commented on src/app/page.tsx:
```diff
@@ -0,0 +1,19 @@
+export default function Home() {
+  return (
+    <>
+      <HeaderSection />
```
very clean i like it!

idiglove commented on src/components/organisms/about-section.tsx:
```diff
@@ -0,0 +1,24 @@
+  return (
+    <SectionWrapper id="about">
+      <ContentWrapper className="flex flex-col text-nlec-green">
+        <SectionTitle className="w-[160px] md:w-[200px] xl:w-[250px]">
```
instead of arbitrary widths. 

use full width then play with padding

idiglove commented on src/components/organisms/contact-us-form.tsx:
```diff
@@ -0,0 +1,147 @@
+
+    emailjs
+      .send("service_g0ej8rl", "template_v3298ya", templateParams, {
+        publicKey: "user_nEmQ6yi4aBPvlCKCkUAfK",
```
use env pls

idiglove commented on src/components/organisms/contact-us-form.tsx:
```diff
@@ -0,0 +1,147 @@
+    };
+
+    emailjs
+      .send("service_g0ej8rl", "template_v3298ya", templateParams, {
```
use env pls

idiglove commented on src/components/organisms/footer.tsx:
```diff
@@ -0,0 +1,99 @@
+  const FOOTER_ITEMS = {
+    links: [
+      {
+        label: "Email",
```
can we extract this info to a json file and under assets

idiglove commented on src/components/organisms/header-section.tsx:
```diff
@@ -0,0 +1,37 @@
+          </div>
+        </div>
+      </div>
+      <div className="relative h-[300px] w-full md:h-[400px] xl:h-[500px]">
```
arbitrary numbers are hard to manage

Noxaelum replied:
I implemented the design by having the top element have negative margins, so the image isn't a position absolute (keeping it bound to the page and having the proper spacing for the bottom section). Modifying the height to be a different style would have to alter the whole implementation

idiglove commented on src/components/organisms/impact-section.tsx:
```diff
@@ -0,0 +1,32 @@
+import ContentWrapper from "../molecules/content-wrapper";
+
+export default function ImpactSection() {
+  const IMPACT_ITEMS = [
```
i would extract to a json under assets. so easy to find where to edit

idiglove commented on src/components/organisms/join-us-section.tsx:
```diff
@@ -0,0 +1,58 @@
+          </SectionTitle>
+          <div className="relative h-1 w-full bg-nlec-green" />
+        </div>
+        <div className="grid gap-4 md:h-[300px] md:grid-cols-3 xl:h-[500px]">
```
try to be not so arbitrary. u will see issues when playing with screen widths

Noxaelum replied:
Percentage values for the height wouldnt work as well as with using a fixed height

idiglove replied:
no need for percentage values. the way to fluid layout works is - the wrapper wraps the content and we add padding. then usually we use max-height intead of determining the height.

i saw problems like this on venture freight, thats why i told you guys to check how i fixed it on my PR

idiglove replied:
to test what ive been saying, play around with the screen widths - mobile, tablet etc

Noxaelum replied:
![image](https://github.com/user-attachments/assets/3e1188a2-436b-4702-9168-1bffbf272507)
Here's what the design looks like using max-width on a tablet width (768px), most likely because the child elements are using `h-full`, but then again moving the fixed height property to the children would be counter-intuitive, as we were already doing that with the current  solution

Noxaelum committed:
Remove comments
Move data to assets
Use percentage values wherever available
Move emailjs keys to env and call using process env
